### PR TITLE
[OSD-16225] : Add clusterrolebinding webhooks to prevent deletion of clusterrolebings in openshift and kube-system namespace

### PR DESCRIPTION
**Pull request by samanthajayasinghe**
Add the webhook for critical clusterrolebinding deletion prevention in following NS 

- openshift-*
- kube-system

### CODE REVIEW

File: pkg/webhooks/clusterrolebinding/clusterrolebinding.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	return fmt.Sprintf(docString, managedNamespaces)
+}
+
+// SyncSetLabelSelector returns the label selector to use in the SyncSet.
+// Return utils.DefaultLabelSelector() to stick with the default
+func (s *ClusterRoleBindingWebHook) SyncSetLabelSelector() metav1.LabelSelector {
+	return utils.DefaultLabelSelector()
+}
+
+func (s *ClusterRoleBindingWebHook) HypershiftEnabled() bool { return true }
```
Comment (bmeng) [inline pkg/webhooks/clusterrolebinding/clusterrolebinding.go:200]: I think you will need to run `make package` to generate package manifests for hypershift

File: pkg/webhooks/clusterrolebinding/clusterrolebinding.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+				APIVersions: []string{"v1"},
+				Resources:   []string{"clusterrolebindings"},
+				Scope:       &scope,
+			},
+		},
+	}
+
+	protectedNamespaces = regexp.MustCompile(managedNamespaces)
+
+	exceptionNamespaces = []string{
```
Comment (bmeng) [inline pkg/webhooks/clusterrolebinding/clusterrolebinding.go:44]: Do we still need to validate the namespace if it is ClusterScope?

**Comment by bmeng** [comment]
/hold 
Hold for now till tomorrow, just to unblock the upcoming promotion, ready for review

File: pkg/webhooks/clusterrolebinding/clusterrolebinding.go (hunk 2 above)
Comment (samanthajayasinghe) [reply pkg/webhooks/clusterrolebinding/clusterrolebinding.go:44]: yeh, `clusterrolebindings` links to the subject  `ServiceAccount` which always comes through a namespace. 
basically, this var allows modifying `clusterrolebindings` which links `exception-Namespaces`

File: pkg/webhooks/clusterrolebinding/clusterrolebinding.go (hunk 1 above)
Comment (samanthajayasinghe) [reply pkg/webhooks/clusterrolebinding/clusterrolebinding.go:200]: Thanks @bmeng , generated the mainfests for HS

Comment (bmeng) [reply pkg/webhooks/clusterrolebinding/clusterrolebinding.go:200]: @mrbarge @ravitri  Any of you know if we have criteria for adding webhook to hypershift clusters?

**Comment by bmeng** [comment]
@samanthajayasinghe could you run the `make docs` to generate the document for the new webhook?

File: pkg/webhooks/clusterrolebinding/clusterrolebinding.go (hunk 1 above)
Comment (mrbarge) [reply pkg/webhooks/clusterrolebinding/clusterrolebinding.go:200]: Nothing documented. I think it's a case-by-case basis of asking the question:
- could a cluster owner even do this action from their hosted cluster? (if no: then not needed)
- if a cluster owner does this action from their hosted cluster, would it potentially bring the cluster into an unsupportable state? (if yes, probably needed)

In this case I think if we want this protection for OSD/ROSA it makes sense to also protect it in hosted clusters.

**Pull request merged**